docs/articles/actors/testing-actor-systems.md

```diff
@@ -3,132 +3,29 @@ uid: testing-actor-systems
+Testkit allows you to test your actors in a controlled but realistic environment. The definition of the environment depends of course very much on the problem at hand and the level at which you intend to test, ranging from simple checks to full system tests.
 
-## Asynchronous Integration Testing with TestKit
-When you are reasonably sure that your actor's business logic is correct, the next step is verifying that it works correctly within its intended environment (if the individual actors are simple enough, possibly because they use the `FSM` module, this might also be the first step). The definition of the environment depends of course very much on the problem at hand and the level at which you intend to test, ranging for functional/integration tests to full system tests. The minimal setup consists of the test procedure, which provides the desired stimuli, the actor under test, and an actor receiving replies. Bigger systems replace the actor under test with a network of actors, apply stimuli at varying injection points and arrange results to be sent from different emission points, but the basic principle stays the same in that a single procedure drives the test.
+The minimal setup consists of the test procedure, which provides the desired stimuli, the actor under test, and an actor receiving replies. Bigger systems replace the actor under test with a network of actors, apply stimuli at varying injection points and arrange results to be sent from different emission points, but the basic principle stays the same in that a single procedure drives the test.
 
 The `TestKit` class contains a collection of tools which makes this common task easy.
 
-```csharp
-public class MyActor : UntypedActor
-{
-    protected override void OnReceive(object message)
-    {
-        if (message.Equals("hello world"))
-        {
-            Sender.Tell(message, Self);
-        }
-        else
-        {
-            Unhandled(message);
-        }
-    }
-
-    public bool TestMe
-    {
-        get { return true; }
-    }
-}
-
-[Fact]
-public void EndBackMessagesUnchanged()
-{
-    var echo = Sys.ActorOf(Props.Create<MyActor>());
-    echo.Tell("hello world");
-    ExpectMsg("hello world");
-}
-```
-The `TestKit` contains an actor named `MyActor` which is the entry point for messages to be examined with the various `ExpectMsg...` assertions detailed below. When mixing in the class `ImplicitSender` this test actor is implicitly used as sender reference when dispatching messages from the test procedure. The `MyActor` may also be passed to other actors as usual, usually subscribing it as notification listener. There is a whole set of examination methods, e.g. receiving all consecutive messages matching certain criteria, receiving a whole sequence of fixed messages or classes, receiving nothing for some time, etc.
+[!code-csharp[IntroSample](../../examples/DocsExamples/Testkit/TestKitSampleTest.cs?range=9-64)]
```

Comment:
Don't use the ranges, use a naming region. Like
`[!code-csharp[WebStoreCustomerFSMActor.cs](../../examples/DocsExamples/Persistence/WebStoreCustomerFSMActor.cs?name=persistent-fsm-domain-events)]`